tests/checker/bad/gold/procedure_map_arg.scilla.gold

```diff
@@ -48,6 +48,25 @@
         "column": 17
       },
       "end_location": { "file": "", "line": 0, "column": 0 }
+    },
+    {
+      "error_message":
+        "Type Bool -> Bool cannot be used as procedure parameter",
```

Comment:
Here is a comment saying that function parameters are allowed: https://github.com/Zilliqa/scilla/blob/master/tests/checker/bad/procedure_map_arg.scilla#L36

Reply:
Yes, hence "We do have testcases for it, though, so it's probably intentional."

What I'm saying is that there isn't really any reason for that restriction, because I'm pretty sure our implementation can handle it. That said, I also don't see any use case for them.